### ISSUE TITLE
`log1p` getting overwritten

### ISSUE THREAD

**Issue by ExpandingMan** [JuliaDiff/DiffRules.jl#14]
There seems to be an extra method from another library [here](https://github.com/JuliaDiff/DiffRules.jl/blob/master/src/rules.jl#L66).  At least on 0.7 this results in the methods for `log1p` getting overwritten.  Would it be ok to either delete this line or, if it's needed for 0.6, move it to within the conditional?

**Comment by jrevels** [comment]
Yes, we should just wrap it in a conditional e.g. 

```julia
if is_version_that_doesnt_need_to_reference_libm(VERSION)
    @define_diffrule Base.log1p(x) = ...
else
    @define_diffrule Base.Math.JuliaLibm.log1p(x) = ...
end
```

I don't know what that predicate is off the top of my head, but it seems like just `Base.log1p` works for v0.7.

**Comment by jrevels** [comment]
closed by #15
